packages/api/src/controllers/taskController.js

```diff
@@ -534,6 +527,11 @@ const taskController = {
   },
 
   async checkCustomDependencies(typeOfTask, data, farm_id) {
+    // TODO: Move this validation to checkCreateTask and checkCompleteTask and apply the middleware to all relevant routes.
+    if ('animals' in data || 'animal_batches' in data) {
+      throw customError(`Invalid field: "animals" or "animal_batches" should not be included.`);
+    }
```

Comment:
You would be able to add animals or batches to any types of tasks by including `animals: [{id: xxx}]` in the API request without this check.